core/src/Kokkos_Array.hpp

```diff
@@ -320,6 +320,21 @@ struct Array<T, KOKKOS_INVALID_INDEX, Array<>::strided> {
       : m_elem(arg_ptr), m_size(arg_size), m_stride(arg_stride) {}
 };
 
+template <typename T, size_t N>
+KOKKOS_INLINE_FUNCTION constexpr bool operator==(const Array<T, N, void>& lhs,
+                                                 const Array<T, N, void>& rhs) {
+  for (int i = 0; i < (int)(lhs.size()); ++i) {
```

Comment:
i would prefer we use the size_type here instead of int

Reply:
yes but that was for casting a var looping over at most Rank , here we are looping over the whole array. 
i realize this in practice will not be an issue, but generally IMO the fewer casts the better. 
ok i think i am fine with this